### PR TITLE
Add API that resolves the native android.app.Fragment instance created for a Page

### PR DESCRIPTION
**Pull request by atanasovg**
The specific use case is to enable the YouTube player plugin in a {N} app. The Google API works either through extending [YouTubeBaseActivity](https://developers.google.com/youtube/android/player/reference/com/google/android/youtube/player/YouTubeBaseActivity) or by creating an instance of the [YouTubePlayerFragment](https://developers.google.com/youtube/android/player/reference/com/google/android/youtube/player/YouTubePlayerFragment) class. Since navigation in the core Modules is fragment-based, it is more naturally to use the second approach of the player API.

This pull adds a method on the `AndroidFrame` interface that returns the native `android.app.Fragment` created for a Page instance. This native Fragment instance is needed so that we can take advantage of the [nested fragments](https://developer.android.com/about/versions/android-4.2.html#NestedFragments) functionality and integrate a YouTube fragment instance.

### CODE REVIEW

File: ui/frame/frame.d.ts [hunk 1]
```diff
@@ -294,6 +294,12 @@ declare module "ui/frame" {
          * Gets or sets whether the page UI will be cached when navigating away from the page.
          */
         cachePagesOnNavigate: boolean;
+
+        /**
+         * Finds the native android.app.Fragment instance created for the specified Page.
+         * @param page The Page instance to search for.
+         */
+        fragmentForPage(page: pages.Page): android.app.Fragment;
```
Comment (hshristov) [inline ui/frame/frame.d.ts]: Don't return native type or clients will need the big andoid.d.ts in order to compile.

File: ui/frame/frame.android.ts [hunk 2]
```diff
@@ -509,6 +509,38 @@ class AndroidFrame extends Observable implements definition.AndroidFrame {
         // can go back only if it is not the main one.
         return this.activity.getIntent().getAction() !== android.content.Intent.ACTION_MAIN;
     }
+    
+    public fragmentForPage(page: pages.Page): android.app.Fragment {
+        if(!page || !this._owner) {
+            return undefined;
+        }
+        
+        var tag;
```
Comment (hshristov) [inline ui/frame/frame.android.ts]: All other fields are defined with `let`

**Comment by hshristov** [comment]
Btw I got another idea. If there is YouTubeActivity can we use `startActivity` method instead of using fragments?

**Comment by atanasovg** [comment]
The approach with the Activity will be far more limited and will require more native code to implement. The approach with Fragments will allow us to create a full-blown plugin that exposes a, say `YouTubeView`, module that will allow us to plug it into our Visual Tree, enabling all the benefits like layouts, data-bindings, styling, etc.

**Comment by atanasovg** [comment]
run ci

**Pull request closed without merging**